### PR TITLE
feat(pdf): derive download filename from content + yyyy-mm-dd suffix (#831)

### PR DESCRIPTION
**Pull request by isamu**
## Summary

PDF downloads in three plugin views (textResponse, wiki, markdown) now produce content-derived filenames with a date suffix instead of generic names like `Assistant.pdf`.

## Items to Confirm / Review

- **AppApi extension** — added `getResultTimestamp(uuid)` to the existing provide/inject contract. Returns the result's creation time from `activeSession.resultTimestamps`, or `undefined` if not in the active session. Plugin views fall back to `Date.now()` when undefined.
- **Local timezone for date** — `formatLocalDate` uses local fields (not UTC). Rationale: the date suffix is user-facing, so it should match the user's wall clock — UTC could produce off-by-one dates near midnight in non-UTC zones.
- **textResponse fallback** — when the response has no extractable title (empty / whitespace-only), the filename becomes `chat-YYYY-MM-DD.pdf`. Confirmed with user.
- **Title cap at 50 chars** — both H1 and first-line paths are truncated. Hard cap; not configurable per plugin yet.
- **Markdown plugin** — kept the existing `filenamePrefix || title || "document"` source; only the date suffix is new.

## User Prompt

> pdfダウンロードするときにファイル名がmulmoclaudeとかassistantって名前になって分かりづらい。その会話、もしくはセッションからファイル名を決め欲しい。日付もあると良いかなname-yyyy-mm-dd

Follow-up clarifications (1問1答):

1. textResponse の名前ソース → **B: 返答自体の最初の行 / H1**
2. 日付ソース → **B: 結果作成時 (resultTimestamps)**
3. textResponse fallback → **B: `chat`**
4. wiki / markdown は今の名前ロジックを残して日付だけ → **OK**

### Helpers

- `buildPdfFilename({ name, fallback, timestampMs })` → `${safeName || fallback}-YYYY-MM-DD.pdf`
- `formatLocalDate(ms)` → `YYYY-MM-DD` in local TZ
- `extractTextResponseTitle(text)` — first `# ` line → first non-empty trimmed line → `""`; both paths capped at 50 chars

### Plumbing

- `useAppApi.getResultTimestamp(uuid: string): number | undefined`
- `App.vue` provides it via `activeSession.value?.resultTimestamps.get(uuid)`

### Plugin views

| Plugin | Before | After |
|---|---|---|
| textResponse | `Assistant.pdf` | `${title-or-first-line}-2026-04-26.pdf` |
| wiki | `MulmoClaude.pdf` | `MulmoClaude-2026-04-26.pdf` |
| markdown | `${filenamePrefix\|title\|document}.pdf` | `${same}-2026-04-26.pdf` |

## Test plan

Manual:

- [ ] Send a chat message → response. Click PDF download.
  - [ ] Response with `# 見出し` → `見出し-2026-04-26.pdf`
  - [ ] Plain response → first line (≤50 chars) + date
  - [ ] Empty edge case → `chat-2026-04-26.pdf`
- [ ] Open a wiki page (e.g. MulmoClaude) → PDF → `MulmoClaude-2026-04-26.pdf`
- [ ] Generate a document via markdown plugin → PDF → `${title}-2026-04-26.pdf`

Automated: 23 new unit tests in `test/utils/files/test_filename.ts` and `test/plugins/textResponse/test_utils.ts` covering happy path, empty / null / undefined, unsafe chars, unicode, truncation, H1 priority, default-timestamp fallback.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * PDF downloads now include the creation date in filenames, formatted as YYYY-MM-DD
  * PDF filenames are intelligently generated from content when available (e.g., titles extracted from responses)
  * Consistent filename formatting across all content types with sensible fallbacks

### CODE REVIEW

File: plans/feat-pdf-filename-from-content.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+### `extractTextResponseTitle`
+
+- `# Hello\n\nbody` → `Hello`
+- `Plain reply text` → `Plain reply text`
+- empty string → `""`
+- whitespace-only → `""`
+- long line → truncated to 50 chars
+- H1 with markdown chars → trimmed
+EOF
```
Comment (isamu) [inline plans/feat-pdf-filename-from-content.md]: Stray `EOF` marker looks like a heredoc artifact from the edit and will render as a literal line in the plan document. Please delete it so the fenced block ends cleanly.

File: src/composables/useAppApi.ts [hunk 2]
```diff
@@ -32,6 +32,13 @@ export interface AppApi {
   startNewChat: (message: string, roleId?: string) => void;
   /** Navigate to a workspace-internal link (wiki page, file, session). */
   navigateToWorkspacePath: (href: string) => void;
+  /**
+   * Look up the creation timestamp (epoch ms) of a tool result in the
```
Comment (isamu) [inline src/composables/useAppApi.ts]: This docstring says `getResultTimestamp()` returns the result's creation time, but the implementation in `App.vue` reads `activeSession.value?.resultTimestamps.get(uuid)`. For restored sessions those timestamps are interpolated from `startedAt`/`updatedAt` rather than persisted per-result times, so the PDF date suffix can be off by a day or more. If the suffix must reflect the true creation date, this needs a persisted timestamp source instead of the interpolated session map.

**Review by isamu — COMMENTED** [review]
Two issues to address before merge.

**Comment by isamu** [comment]
CODEX VERDICT: CHANGES REQUESTED
- `plans/feat-pdf-filename-from-content.md` still contains a literal `EOF` line at the end of the fenced block.
- `src/composables/useAppApi.ts` documents `getResultTimestamp()` as a creation-time lookup, but restored sessions only have interpolated timestamps, so the new PDF date suffix can be inaccurate for older sessions.

**Review by isamu — COMMENTED** [review]
CODEX VERDICT: LGTM
- Confirmed the stray heredoc EOF marker is removed from `plans/feat-pdf-filename-from-content.md`.
- Confirmed `src/composables/useAppApi.ts` now describes `getResultTimestamp()` accurately for restored sessions, matching the `activeSession.resultTimestamps` plumbing in `src/App.vue`.
- Rechecked the PDF filename flow in `textResponse`, `wiki`, and `markdown`; the date suffix now comes from the result timestamp when available and falls back cleanly.
- No additional correctness, security, accessibility, or i18n issues found in this pass.
- I attempted to run the new unit tests, but the sandbox blocked the TSX IPC pipe setup, so test execution could not complete here.
